roaring/roaring.go

```diff
@@ -2166,40 +2144,42 @@ func (c *Container) arrayWriteTo(w io.Writer) (n int64, err error) {
 	//}
 
 	// Write sizeof(uint16) * cardinality bytes.
-	nn, err := w.Write((*[0xFFFFFFF]byte)(unsafe.Pointer(&c.array[0]))[:2*c.n])
+	nn, err := w.Write((*[0xFFFFFFF]byte)(unsafe.Pointer(&array[0]))[: 2*c.n : 2*c.n])
```

Comment:
can you explain the significance of adding the capacity to the slicing op?

My read is that we previously weren't doing this, and so technically the slice's capacity value was 0xFFFFFFF or whatever. This didn't matter because all the containers were mapped, and so we weren't appending to the slices and thereby using capacity we didn't actually have.

Reply:
Yup. So, it's harmless right now, but when I was doing the packed representation, storing that unreasonably large value in capacities resulted in overwriting other bits of the representation. But I'd also like to be a little insured against the compiler going insane and deciding that it can do some weird optimization because it "knows" that the capacity of the slice is large.